src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftAPIClient.java

```diff
@@ -2,194 +2,134 @@
-    private String userId;
-    private String userPassword;
-    private String tokenAuthUrl;
-    private String swiftUrl;
-    private String token;
+    private final String swiftUrl;
+    private final String tokenAuthUrl;
+    private final String projectId;
+    private final String userId;
+    private final String userPassword;
 
-    public static final Logger LOGGER = Logger.getLogger(SwiftAPIClient.class);
+    private IdentityToken token;
 
-    public SwiftAPIClient(Properties properties) throws SwiftPermanentStorageException {
+    SwiftAPIClient(Properties properties)
+        throws InvalidPropertyException, IOException, JSONException {
         if (!checkProperties(properties))
-            throw new SwiftPermanentStorageException("Error on validate the file. Missing properties for start Swift API Client.");
-
-        this.properties = properties;
-
-        projectId = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_PROJECT_ID);
-        userId = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_USER_ID);
-        userPassword = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_PASSWORD);
-        tokenAuthUrl = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_AUTH_URL);
-        swiftUrl = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_SWIFT_URL);
-
-        handleTokenUpdate(Executors.newScheduledThreadPool(1));
-
-        LOGGER.info("Waiting to get token...");
-        while (token == null) {
-            try {
-                Thread.sleep(1000);
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-        }
+            throw new InvalidPropertyException("Error on validate the file. Missing properties for start Swift API Client.");
+        this.swiftUrl = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_SWIFT_URL);
+        this.tokenAuthUrl = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_AUTH_URL);
+        this.projectId = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_PROJECT_ID);
+        this.userId = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_USER_ID);
+        this.userPassword = properties.getProperty(SapsPropertiesConstants.FOGBOW_KEYSTONEV3_PASSWORD);
+        this.token = KeystoneV3IdentityRequestHelper.createIdentityToken(tokenAuthUrl, projectId, userId, userPassword);
     }
 
     private boolean checkProperties(Properties properties) {
         String[] propertiesSet = {
-                SapsPropertiesConstants.FOGBOW_KEYSTONEV3_PROJECT_ID,
-                SapsPropertiesConstants.FOGBOW_KEYSTONEV3_USER_ID,
-                SapsPropertiesConstants.FOGBOW_KEYSTONEV3_PASSWORD,
-                SapsPropertiesConstants.FOGBOW_KEYSTONEV3_AUTH_URL,
-                SapsPropertiesConstants.FOGBOW_KEYSTONEV3_SWIFT_URL
+            SapsPropertiesConstants.FOGBOW_KEYSTONEV3_SWIFT_URL,
```

Comment:
Uhm,  should we still have anything named FOGBOW in this PR?

Reply:
These constants will be corrected in #98 .

Reply:
ok.